src/box/memtx_tree.cc

```diff
@@ -231,6 +232,17 @@ struct tree_iterator {
 	enum iterator_type type;
 	struct memtx_tree_key_data<USE_HINT> key_data;
 	struct memtx_tree_data<USE_HINT> current;
+	/**
+	 * For functional indexes only: copy of the functional index key at the
+	 * current iterator position. Allocated from current_func_key_buf or on
+	 * malloc.
+	 *
+	 * Since pinning a tuple doesn't prevent its functional keys from being
+	 * deleted, we need to copy it so that we can use it to restore the
+	 * iterator position.
+	 */
+	void *current_func_key;
+	char current_func_key_buf[128];
```

Comment:
Please make it no more that 32 bytes. I would make 16. It's enough for numbers and string words.

Reply:
Set to 32 bytes.